### PR TITLE
Added an example on WebSocketProtocolHandshakeHandler usage

### PR DESCRIPTION
**Pull request by altmind**
I had some hard times finding an example on how to enable websocket per-message deflate.
I've added an extra example highliting usage of WebSocketProtocolHandshakeHandler and PerMessageDeflateHandshake in this PR.

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
Can one of the admins verify this patch?

**Comment by stuartwdouglas** [comment]
This is ok to test

File: examples/src/main/java/io/undertow/examples/websockets_extension/WebSocketServer.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import io.undertow.websockets.extensions.PerMessageDeflateHandshake;
+import io.undertow.websockets.spi.WebSocketHttpExchange;
+
+import static io.undertow.Handlers.path;
+import static io.undertow.Handlers.resource;
+
+/**
+ * @author Stuart Douglas
+ */
+@UndertowExample("Web Sockets")
```
Comment (stuartwdouglas) [inline examples/src/main/java/io/undertow/examples/websockets_extension/WebSocketServer.java]: There is already an example with the same name, this should be called 'Web Socket Extensions'

**Comment by undertow-pull-request** [comment]
**Windows** [Build 2459](https://ci.wildfly.org/viewLog.html?buildId=107148&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of ac2d42c740449617c51d17e7632ccedd0e55d804
Summary: Execution timeout; tests passed: 1815, ignored: 611 Build time: 00:45:14

**Pull request merged**